src/js/carousel.js

```diff
@@ -0,0 +1,33 @@
+const carousel = document.querySelector(".carousel");
+const slider = carousel.querySelector(".slider");
+const dotsContainer = carousel.querySelector(".dots-list");
+const dots = Array.from(dotsContainer.children);
+const numberOfPages = dots.length
+console.log(numberOfPages)
+
+let currentIndex = 0;
+
+
+// auto-play carousel logic
+setInterval(() => {
+  dots[currentIndex].classList.remove("active");
+  currentIndex = currentIndex === numberOfPages - 1 ? 0 : currentIndex + 1;
+  slider.style.transform = `translateX(${-100 * currentIndex}%)`;
+  slider.style.transition = ".7s";
```

Comment:
you can avoid adding this inline style with js. Pleas add on _carousel.scss_ `transition: transform .7s ease; ` for the .slider selector.